tc/core/polyhedral/cuda/codegen.cc

```diff
@@ -39,6 +39,39 @@ namespace polyhedral {
 
 namespace {
 
+static std::string halideTypeString(const Halide::Type& t) {
+  if (t.is_bool()) {
+    return "bool";
+  } else if (t.is_int() && t.bits() == 8) {
+    return "char";
+  } else if (t.is_int() && t.bits() == 16) {
+    return "short";
+  } else if (t.is_int() && t.bits() == 32) {
+    return "int";
+  } else if (t.is_int() && t.bits() == 64) {
+    return "long";
+  } else if (t.is_uint() && t.bits() == 8) {
+    return "unsigned char";
+  } else if (t.is_uint() && t.bits() == 16) {
+    return "unsigned short";
+  } else if (t.is_uint() && t.bits() == 32) {
+    return "unsigned int";
+  } else if (t.is_uint() && t.bits() == 64) {
+    return "unsigned long";
+  } else if (t.is_float() && t.bits() == 16) {
+    return "half";
+  } else if (t.is_float() && t.bits() == 32) {
+    return "float";
+  } else if (t.is_float() && t.bits() == 64) {
+    return "double";
+  }
+  LOG(ERROR) << "t.is_int(): " << t.is_int();
+  LOG(ERROR) << "t.bits(): " << t.bits();
```

Comment:
don't we want this to crash here rather than return some `nameBitsize` string?

Reply:
not clear, will see once we get vector types in, if I can avoid bloating this function that would be better

Reply:
the print itself is spurious though